src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -0,0 +1,140 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SpectralManager : ISpectralManager
+{
+    private const string SpectralVersion = "6.11.0";
+
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiToolsDirectoryPath;
+    private readonly string _toolDirectory;
+
+    public SpectralManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiToolsDirectoryPath = openApiToolsDirectoryPath;
+        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, $"spectral/{SpectralVersion}");
+    }
+
+    public async Task InstallSpectralAsync(CancellationToken cancellationToken)
+    {
+        this.CreateRequiredDirectories();
+        var executableFileName = this.GetSpectralFileName();
+
+        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{executableFileName}";
+
+        await this.DownloadFileAsync(url, $"{this._toolDirectory}/{executableFileName}", cancellationToken);
+    }
+
+    private void CreateRequiredDirectories()
+    {
+        Directory.CreateDirectory(Path.Combine(this._openApiToolsDirectoryPath, "spectral"));
+        Directory.CreateDirectory(this._toolDirectory);
+    }
+
+    private string GetSpectralFileName()
+    {
+        var osType = GetOperatingSystem();
+        var architecture = GetArchitecture();
+
+        if (osType == "linux")
+        {
+            var distro = File.Exists("/etc/os-release") ? File.ReadAllText("/etc/os-release") : string.Empty;
+            if (distro.Contains("Alpine Linux"))
+            {
+                osType = "alpine";
+                this._loggerWrapper.Helper.LogMessage(MessageImportance.Low, "Installing on Alpine Linux.");
+            }
+        }
+
+        var fileName = $"spectral-{osType}-{architecture}";
+
+        if (osType == "win")
+        {
+            fileName = "spectral.exe";
+            this._loggerWrapper.Helper.LogMessage(MessageImportance.Low, "Installing on Windows.");
+        }
+
+        return fileName;
+    }
+
+    private async Task DownloadFileAsync(string url, string destination, CancellationToken cancellationToken)
+    {
+        if (File.Exists(destination))
```

Comment:
Here a "partial file" could exist. Like, the download could fail but we would still have written bytes to the file system, resulting in an unusable executable and we would never try to download it again.

Idea for later => maybe we could do a checksum validation to ensure that the binary is properly downloaded

Reply:
This could be a volontary debt to not manage that edge case, it's easy to fix (delete folder)